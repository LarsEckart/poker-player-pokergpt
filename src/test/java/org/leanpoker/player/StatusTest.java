package org.leanpoker.player;

import org.approvaltests.combinations.CombinationApprovals;
import org.junit.jupiter.api.Test;

class StatusTest {

    private static String call(Integer ourIndex, Integer dealerIndex) {

        Status status = GameState.getStatus(ourIndex, dealerIndex, 4);

        String result = "";
        if (dealerIndex == 0) {
            result += "D|s|B|1\n";
        } else if (dealerIndex == 1) {
            result += "1|D|s|B\n";
        } else if (dealerIndex == 2) {
            result += "B|1|D|s\n";
        } else if (dealerIndex == 3) {
            result += "s|B|1|D\n";
        }

        return status.biddingPosition() + " -> " + result;
    }

    @Test
    void name() {
        Integer[] places = {0, 1, 2, 3};
        CombinationApprovals.verifyAllCombinations(
                StatusTest::call,
                places,
                places
        );
    }
}
