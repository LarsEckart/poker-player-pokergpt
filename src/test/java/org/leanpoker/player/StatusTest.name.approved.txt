[0, 0] => 2 -> D|s|B|1
 
[0, 1] => 1 -> 1|D|s|B
 
[0, 2] => 4 -> B|1|D|s
 
[0, 3] => 3 -> s|B|1|D
 
[1, 0] => 3 -> D|s|B|1
 
[1, 1] => 2 -> 1|D|s|B
 
[1, 2] => 1 -> B|1|D|s
 
[1, 3] => 4 -> s|B|1|D
 
[2, 0] => 4 -> D|s|B|1
 
[2, 1] => 3 -> 1|D|s|B
 
[2, 2] => 2 -> B|1|D|s
 
[2, 3] => 1 -> s|B|1|D
 
[3, 0] => 1 -> D|s|B|1
 
[3, 1] => 4 -> 1|D|s|B
 
[3, 2] => 3 -> B|1|D|s
 
[3, 3] => 2 -> s|B|1|D
 
